remote-storage-managers/hdfs/src/main/java/org/apache/kafka/rsm/hdfs/HDFSRemoteStorageManager.java

```diff
@@ -115,12 +115,12 @@ public InputStream fetchTimestampIndex(RemoteLogSegmentMetadata remoteLogSegment
     }
 
     @Override
-    public boolean deleteLogSegment(RemoteLogSegmentMetadata remoteLogSegmentMetadata) throws RemoteStorageException {
+    public void deleteLogSegment(RemoteLogSegmentMetadata remoteLogSegmentMetadata) throws RemoteStorageException {
         try {
             String path = getSegmentRemoteDir(remoteLogSegmentMetadata.remoteLogSegmentId());
 
             FileSystem fs = getFS();
-            return fs.delete(new Path(path), true);
+            fs.delete(new Path(path), true);
```

Comment:
Return value should be handled here to throw an Exception and wanted to add more javadoc around failures.